### DISCUSSION TITLE
New setting for auto backup of mcc file

### DISCUSSION THREAD

**Discussion by JaimeLeon2** [MobiFlight/MobiFlight-Connector#687] [Ideas]
**Is your feature request related to a problem? Please describe.**
There is no auto backup function in Mobiflight allowing the user to permanently overwrite this "good" config with a "bad" one.  Once saved, the user cannot go back.

**Describe the solution you'd like**
Create a new setting in the Settings Menu for Auto backup.  When a Save file command is given and this option is checked, Mobiflight will save the previous config to a BK0 to BK9 file, before saving the current config to the mcc file.  Up to 10 backups can be saved this way.  Mobiflight decides which is the next BKn file by checking the one with the latest date and using the next number.

Extra credit if Mobiflight can restore the mcc config from an existing backup file.

Same logic could be extended to the mfmc files.

**Describe alternatives you've considered**
No good alternatives exist.  User can create the backup by changing the name or copying the mcc files in Windows Explorer.  Both options are far from being automatic.

**Additional context**
none.

**Comment by DocMoebiuz** [comment]
I moved this into discussions to see what other users think about this.

**Reply by pizman** [reply]
Basically i interesting idea....
But like in other points.... Do we need to protect a user from every possible problem ( he did by his on fault) ?

I ask myself.... Also a professional Software like Excel, Word or so not have this and not think about this!
If i got my Doctoral Thesis as a *.doc word document....
I open it.... Mark 20 pages of text and press "DEL" and then i´m stupid enough to SAVE and Confirm it....
YES   Then its gone !   No Backup  No protection.

MF do the same....  If a user "overwrite" by intent or active confirm the Question " Do you want to Save" .....
Then it is like it is !

Summary:   I think it would be cool and usefull....   But its no "must have" for me.   Users need to take care what they are doing themself!
Before i change a File, i simply make a "SaveAs" myself.... So i got a backup.
